### PR TITLE
2697: Optimize gradle builds

### PR DESCRIPTION
**Pull request by ztefanie**
### Short description
Improvments for gradle builds

### Proposed changes
- Enable parallel build
- Disable hermes undefined warnings

See scan before here: https://scans.gradle.com/s/iquknvkkj7wjg
See scan after here: https://scans.gradle.com/s/jjay2v7tbxmvw

### Side effects

- gradle builds in the CI 

### Resolved issues

Fixes: #2697 

---

### CODE REVIEW

**Review by steffenkleinle — APPROVED** [review]
LGTM

File: native/android/app/build.gradle [hunk 1]
```diff
@@ -55,6 +55,7 @@ react {
     //
     //   The list of flags to pass to the Hermes compiler. By default is "-O", "-output-source-map"
     // hermesFlags = ["-O", "-output-source-map"]
+    hermesFlags =  ["-O", "-output-source-map", "-Wundefined-variable"] // this will disable all undefined warnings (they are created due to hermes ahead of time compilation)
```
Comment (f1sh1918) [inline native/android/app/build.gradle:58]: you may remove the comment above

File: native/android/gradle.properties [hunk 2]
```diff
@@ -37,4 +37,6 @@ newArchEnabled=false
 
 # Use this property to enable or disable the Hermes JS engine.
 # If set to false, you will be using JSC instead.
-hermesEnabled=true
\ No newline at end of file
+hermesEnabled=true
+
+org.gradle.parallel=false
```
Comment (f1sh1918) [inline native/android/gradle.properties:42]: should this really be false?
Due to your description parallel builds are enabled with this pr.
Then its not very intuitive to enable sth by setting it false. This just confuses me

**Review by f1sh1918 — COMMENTED** [review]
Build time was really good. Nice work! 👍 
I'm just a little bit confused about the `parallel` flag. Check comment :)

File: native/android/gradle.properties (hunk 2 above)
Comment (ztefanie) [reply native/android/gradle.properties:42]: You are right, I messed up, will set it to true.

**Review by f1sh1918 — APPROVED** [review]
looks good

**Pull request merged**